frontend/src/app/components/search-results/search-results.component.ts

```diff
@@ -18,18 +19,19 @@ export class SearchResultsComponent implements OnInit {
   searchForm: FormGroup;
 
   constructor(private userService: UserService, 
-              private apiService: APIService, 
+              private accountService: AccountService,
+              private searchService: SearchService, 
               private router: Router, 
               private route: ActivatedRoute,
               private snackBar: MatSnackBar,
               private fb: FormBuilder) { }
 
   ngOnInit(): void {
-    if (this.userService.isAccountLoggedIn()) {
```

Comment:
There are a few instances here where I take the Form Builder out of the if statements on some ngOnInits, this is because sometimes errors were thrown in the chrome debugger for valid reasons